src/main/java/fi/aalto/cs/apluscourses/presentation/base/TreeViewModel.java

```diff
@@ -1,7 +1,9 @@
 package fi.aalto.cs.apluscourses.presentation.base;
 
-import javax.swing.tree.TreeModel;
+import java.util.List;
+import org.jetbrains.annotations.Nullable;
 
 public interface TreeViewModel {
-  TreeModel toTreeModel();
+  @Nullable
+  List<? extends TreeViewModel> getSubtrees();
```

Comment:
This is a very nice solution and also takes the `DefaultMutableTreeNode` stuff out of the view models :+1: To nit: perhaps `getChildren` is a more descriptive name?